iceoryx_posh/include/iceoryx_posh/capro/service_description.hpp

```diff
@@ -133,13 +131,9 @@ class ServiceDescription
     /// @brief Returns the scope of a ServiceDescription
     Scope getScope() const noexcept;
 
-    /// @brief Returns true for valid ServiceDescription
-    /// false for ServiceDescription that contain InvalidStrings.
-    /// @return bool, true if ServiceDescription is valid, false otherwise
-    bool isValid() const noexcept;
 
     ///@{
-    /// Getters for the integer and string IDs
+    /// Getters for the string IDs
     IdString_t getServiceIDString() const noexcept;
```

Comment:
We still return by value here. This is bad for performance everywhere it is used. I would prefer to have it return a `const IdString_t&`. Furthermore I am still not sure whether in a strict camel case model it would not be `FooIdString`, but I do not care much.

 This might break the usage somewhere, so in the future an alternative would be to have functions like `const IdString_t& serviceIdString()` or just `const IdString_t& service()`.